test/gurka/test_access.cc

```diff
@@ -0,0 +1,621 @@
+      A----B----C----D------------E
+           |         |            |
+           F         G            H
+           |         |            |
+           I----J----K------------L
+  )";
+
+  const gurka::ways ways = {
+      {"AB", {{"highway", "residential"}}}, {"BC", {{"highway", "residential"}}},
+      {"CD", {{"highway", "residential"}}}, {"DE", {{"highway", "residential"}}},
+      {"BF", {{"highway", "residential"}}}, {"DG", {{"highway", "residential"}}},
+      {"EH", {{"highway", "residential"}}}, {"FI", {{"highway", "residential"}}},
+      {"GK", {{"highway", "residential"}}}, {"HL", {{"highway", "residential"}}},
+      {"IJ", {{"highway", "residential"}}}, {"JK", {{"highway", "residential"}}},
+      {"KL", {{"highway", "residential"}}},
+  };
+
+  const gurka::nodes nodes = {{"F", {{"motor_vehicle", "no"}}}, {"G", {{"motorcar", "no"}}}};
+
+  const auto layout = gurka::detail::map_to_coordinates(ascii_map, 100);
+  auto map = gurka::buildtiles(layout, ways, nodes, {}, "test/data/gurka_node_access");
+
+  for (auto& c : costing) {
+    if (c == "auto" || c == "taxi")
+      validate_path(gurka::do_action(valhalla::Options::route, map, {"A", "I"}, c),
+                    {"AB", "BC", "CD", "DE", "EH", "HL", "KL", "JK", "IJ"});
+    else if (c == "bicycle" || c == "pedestrian")
+      validate_path(gurka::do_action(valhalla::Options::route, map, {"A", "I"}, c),
+                    {"AB", "BF", "FI"});
+    else
+      validate_path(gurka::do_action(valhalla::Options::route, map, {"A", "I"}, c),
+                    {"AB", "BC", "CD", "DG", "GK", "JK", "IJ"});
+  }
+}
+
+TEST(Standalone, RouteOnPrivateAccess) {
+  constexpr double gridsize_metres = 10;
+
+  const std::string ascii_map = R"(
+        A---B---C---D
+            |   |   |
+            E   F   G
+    )";
+
+  const gurka::ways ways = {
+      {"AB", {{"highway", "primary"}}},
+      {"BC", {{"highway", "primary"}}},
+      {"CD", {{"highway", "primary"}}},
+      {"BE", {{"highway", "service"}, {"access", "private"}}},
+      {"CF", {{"highway", "service"}, {"access", "private"}, {"service", "driveway"}}},
+      {"DG", {{"highway", "service"}, {"access", "private"}, {"service", "parking_aisle"}}},
+  };
+
+  const auto layout =
+      gurka::detail::map_to_coordinates(ascii_map, gridsize_metres, {5.1079374, 52.0887174});
+  auto map = gurka::buildtiles(layout, ways, {}, {}, "test/data/gurka_route_on_private_access",
+                               build_config);
+
+  for (auto& c : costing) {
+    auto result = gurka::do_action(valhalla::Options::route, map, {"A", "E"}, c);
+    gurka::assert::raw::expect_path(result, {"AB", "BE"});
+
+    result = gurka::do_action(valhalla::Options::route, map, {"A", "F"}, c);
+    gurka::assert::raw::expect_path(result, {"AB", "BC", "CF"});
+
+    result = gurka::do_action(valhalla::Options::route, map, {"A", "G"}, c);
+    gurka::assert::raw::expect_path(result, {"AB", "BC", "CD", "DG"});
+  }
+}
```

Comment:
these were all existing tests. i hate that we have so many little one off tests. it makes the CI build slow and it also makes stuff hard to find. all of these tests are verifying stuff with access, which is a big subject area in the code base of course. to me, its better to just have all the access crap in one test. i know that now its a big test but i think its better than 7 different executables that all need to be run (when we already have hundreds)